### PR TITLE
chore(functions): #1195 #1187 refactored cloud functions

### LINKED ISSUE #1187: Update cloud functions tslint to match angular (strict)

### LINKED ISSUE #1195: Split up cloud functions into smaller files

### LINKED ISSUE #1211: Ignore .history folder in tsconfig
Auto complete is picking up `.history` folder and should be ignored

### LINKED ISSUE #1213: Move http client from Axios to Request

### LINKED ISSUE #1215: Upgrade firebase functions to node v8

### PR DESCRIPTION
**Pull request by eddiejaoude**
closes #1211 
closes #1195 
closes #1187 
closes #1213 
closes #1215 

### Notes

A summary of what was achieved in this PR

- [x] tsconfig updated to ignore `.history`
- [x] cloud functions to use strict types
- [x] split cloud functions in to multiple files
- [x] renamed functions to be consistent
- [x] updated node from v6 to v8

Note: `any` was used in a few places as could not find a suitable type

### CODE REVIEW

File: functions/src/repository/info.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+export const getRepositoryInfo: functions.HttpsFunction = functions.https.onCall((input: Input, context: CallableContext) => {
+    return axios.all([
+        GitHubClient(input.token).get(`/repos/${input.fullName}`),
+        GitHubClient(input.token).get(`/repos/${input.fullName}/pulls?state=open`),
+        GitHubClient(input.token).get(`/repos/${input.fullName}/events`),
+        GitHubClient(input.token).get(`/repos/${input.fullName}/releases`),
+    ])
+    .then(axios.spread((repository: GitHubResponse<GitHubRepositoryInput>, pullrequests: GitHubResponse<GitHubPullRequestInput[]>, events: GitHubResponse<GitHubEventInput[]>, releases: GitHubResponse<GitHubReleaseInput[]>) => admin
```
Comment (eddiejaoude) [inline functions/src/repository/info.ts]: `npm run build` error

```shell
src/repository/info.ts(22,24): error TS2345: Argument of type '(repository: GitHubResponse<GitHubRepositoryInput>, pullrequests: GitHubResponse<GitHubPullReques...' is not assignable to parameter of type '(...args: GitHubResponse<GitHubRepositoryInput>[]) => Promise<WriteResult>'.
  Types of parameters 'pullrequests' and 'args' are incompatible.
    Type 'GitHubResponse<GitHubRepositoryInput>' is not assignable to type 'GitHubResponse<GitHubPullRequestInput[]>'.
      Type 'GitHubRepositoryInput' is not assignable to type 'GitHubPullRequestInput[]'.
        Property 'length' is missing in type 'GitHubRepositoryInput'.
```

File: functions/src/mappers/github/organisation.mapper.ts [hunk 2]
```diff
@@ -1,5 +1,17 @@
+
```
Comment (webkhushboo) [inline functions/src/mappers/github/organisation.mapper.ts]: Please remove extra lines before interface declaration.

File: functions/src/mappers/github/organisation.mapper.ts [hunk 3]
```diff
@@ -1,5 +1,17 @@
...
+
+export interface GitHubOrganisationtInput {
+    login: string;
+    avatar_url: string;
+}
+
```
Comment (webkhushboo) [inline functions/src/mappers/github/organisation.mapper.ts]: Please remove extra lines before interface declaration.

File: functions/src/mappers/github/organisation.mapper.ts (hunk 2 above)
Comment (eddiejaoude) [reply functions/src/mappers/github/organisation.mapper.ts]: Good spot 👍

File: functions/src/mappers/github/organisation.mapper.ts (hunk 3 above)
Comment (eddiejaoude) [reply functions/src/mappers/github/organisation.mapper.ts]: Good spot 👍

**Comment by eddiejaoude** [comment]
@webkhushboo I made the changes from your comment, but the PR will need approving again, thank you 🤓

**Pull request merged**